src/components/ActionSheetAwareScrollView/ActionSheetKeyboardSpace.js

```diff
@@ -0,0 +1,237 @@
+import React, {
+    useContext, useEffect, useMemo,
+} from 'react';
+import {Dimensions, Keyboard} from 'react-native';
+import Reanimated, {
+    useWorkletCallback,
+    Easing,
+    KeyboardState,
+    useAnimatedKeyboard,
+    useAnimatedStyle, useDerivedValue, withTiming,
+} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import styles from '../../styles/styles';
+import {Actions, States, ActionSheetAwareScrollViewContext} from './ActionSheetAwareScrollViewContext';
+
+const config = {
+    duration: 350,
+    easing: Easing.bezier(0.33, 0.01, 0, 1),
+};
+
+function ActionSheetKeyboardSpace(props) {
+    const safeArea = useSafeAreaInsets();
+    const keyboard = useAnimatedKeyboard();
+
+    // similar to using `global` in worklet but it's just a local object
+    const ctx = useMemo(() => ({}), []);
+    const windowHeight = Dimensions.get('screen').height;
+    const {
+        currentActionSheetState, transitionActionSheetStateWorklet: transition, transitionActionSheetState, resetStateMachine,
+    } = useContext(ActionSheetAwareScrollViewContext);
+
+    useEffect(() => {
+        // sometimes it might trigger multiple times the same actions
+        let lastAction = null;
+        const onDidShow = () => {
```

Comment:
Minor detail, but one of our review guidelines is that "functions should be named for what they do, not what they handle". So I would recommend just declaring this inline in the call to `Keyboard.addListener`. I think that's a bit easier to understand why we have these nested functions too.

Reply:
Removed them since we're not gonna use it anymore